api/v1alpha1/const.go

```diff
@@ -24,6 +24,8 @@ const (
 	ExtendedDaemonSetCanaryPausedReasonAnnotationKey = "extendeddaemonset.datadoghq.com/canary-paused-reason"
 	// ExtendedDaemonSetCanaryFailedAnnotationKey annotation key used on ExtendedDaemonset in order to detect if a canary deployment has failed.
 	ExtendedDaemonSetCanaryFailedAnnotationKey = "extendeddaemonset.datadoghq.com/canary-failed"
+	// ExtendedDaemonSetCanaryFailedReasonAnnotationKey annotation key used on ExtendedDaemonset in order to provide a reason that the a canary deployment is failed.
+	ExtendedDaemonSetCanaryFailedReasonAnnotationKey = "extendeddaemonset.datadoghq.com/canary-failed-reason"
```

Comment:
should we put the reason in the status instead?

Reply:
We still need to keep these annotations because they are at the EDS level and if something goes wrong with the ERS we won't be able to tell, besides we have a command to fail the canary via the annotation. For now I added `canary-failed` status of the canary to keep track of it in the status, but it's a transient state as it will be deleted shortly after that.